src/models/User.js

```diff
@@ -1,55 +0,0 @@
-const mongoose = require("mongoose");
-const uniqueValidator = require("mongoose-unique-validator");
-const crypto = require("crypto");
-const { secret } = require("../config");
-
-const UserSchema = new mongoose.Schema(
-  {
-    username: {
-      type: String,
-      lowercase: true,
-      unique: true,
-      required: [true, "can't be blank"],
-      match: [/^[a-zA-Z0-9]+$/, "is invalid"],
-      index: true,
-    },
-    email: {
-      type: String,
-      lowercase: true,
-      unique: true,
-      required: [true, "can't be blank"],
-      match: [/\S+@\S+\.\S+/, "is invalid"],
-      index: true,
-    },
-    bio: String,
-    image: String,
-    favorites: [{ type: mongoose.Schema.Types.ObjectId, ref: "Article" }],
-    following: [{ type: mongoose.Schema.Types.ObjectId, ref: "User" }],
-    hash: String,
-    salt: String,
-  },
-  { timestamps: true }
-);
-
-UserSchema.plugin(uniqueValidator, { message: "is already taken." });
-
-UserSchema.methods.validPassword = (password) => {
-  const hash = crypto
-    .pbkdf2Sync(password, this.salt, 10000, 512, "sha512")
-    .toString("hex");
-  return this.hash === hash;
-};
-
-UserSchema.methods.setPassword = (password) => {
-  this.salt = crypto.randomBytes(16).toString("hex");
-  this.hash = crypto
-    .pbkdf2Sync(password, this.salt, 10000, 512, "sha512")
-    .toString("hex");
-};
-
-UserSchema.methods.toAuthJSON = () => ({
-  username: this.username,
-  email: this.email,
-});
-
-mongoose.model("User", UserSchema);
```

Comment:
I thought some of these were to be used as a reference. Won't it be an issue if we remove them?

Reply:
UserSchema has reference to mongoose model. That is the reason I deleted all of them

Reply:
No. All these are related to mongoose which is a no-SQL database and we are going to use Postgres